crates/cheatcodes/src/context.rs

```diff
@@ -0,0 +1,75 @@
+//! Implementations of [`Context`](crate::Group::Context) cheatcodes.
+
+use crate::{Cheatcode, Cheatcodes, Result, Vm::*};
+use alloy_sol_types::SolValue;
+use once_cell::sync::OnceCell;
+use std::mem::discriminant;
+
+/// Stores the forge execution context for the duration of the program.
+static FORGE_CONTEXT: OnceCell<ForgeContext> = OnceCell::new();
+
+/// Possible forge execution contexts.
+pub enum ForgeContext {
```

Comment:
We probably also should have:
- `Script` which equals `ScriptDryRun || ScriptBroadcast ScriptResume` (i.e. any script mode)
- Another context that returns if you are in any test context, i.e. `Test || Coverage || Snapshot`. I'm stuck on naming here though. Perhaps `Test == StandardTest || Coverage || Snapshot`?

Reply:
ah, yep, good idea, will add `Script` and `Test` for any script / test mode rename regular test context to `StandardTest`

Reply:
actually there's no need to define new contexts, will just use existing and add `isScriptContext` and `isStandardTestContext`

Reply:
changed, please check updated PR decription